src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVFastqUtils.java

```diff
@@ -4,49 +4,245 @@
+                cigar = null;
+            } else {
+                contig = read.getContig();
+                start = read.getStart();
+                forwardStrand = !read.isReverseStrand();
+                cigar = read.getCigar();
+            }
+        }
+
+        public Mapping(final String mappingDescription) {
+            Utils.nonNull(mappingDescription);
+            if (mappingDescription.equals(UNMAPPED_DESCRIPTION_STR)) {
+                contig = null;
+                start = -1;
+                forwardStrand = true;
+                cigar = null;
+            } else {
+                final Matcher matcher = MAPPING_DESCRIPTION_PATTERN.matcher(mappingDescription);
+                if (!matcher.find()) {
+                    throw new IllegalArgumentException("invalid mapping description: '" + mappingDescription + "'");
+                } else {
+                    contig = matcher.group(1);
+                    start = Integer.parseInt(matcher.group(2));
+                    cigar = TextCigarCodec.decode(matcher.group(4));
+                    forwardStrand = Strand.decode(matcher.group(3)) == Strand.POSITIVE;
+                }
+            }
+        }
+
+        public boolean isMapped() {
+            return contig != null;
+        }
+
+        @Override
+        public String getContig() {
+            return contig;
+        }
+
+        @Override
+        public int getStart() {
+            return start;
+        }
+
+        @Override
+        public int getEnd() {
+            return start;
+        }
+
+        public Cigar getCigar() {
+            return cigar;
+        }
+
+        public boolean isForwardStrand() {
+            return forwardStrand;
+        }
+
+        public String toString() {
+            if (isMapped()) {
+                return MAPPING_FIELD_EQUAL_TO + String.join(HEADER_FIELD_ARRAY_SEPARATOR_STR, getContig() + SimpleInterval.CONTIG_SEPARATOR + getStart(), (forwardStrand ? Strand.POSITIVE : Strand.NEGATIVE).encodeAsString(), cigar.toString());
+            } else {
+                return UNMAPPED_DESCRIPTION_STR;
+            }
+        }
+    }
+
     @DefaultSerializer(FastqRead.Serializer.class)
     public static final class FastqRead implements FermiLiteAssembler.BasesAndQuals {
-        private final String name;
+        private final String header;
         private final byte[] bases;
         private final byte[] quals;
 
-        public FastqRead( final String name, final byte[] bases, final byte[] quals ) {
-            this.name = name;
+        public FastqRead( final GATKRead read ) {
+            this(Utils.nonNull(read).getName(),
+                 read.isPaired() ? OptionalInt.of(read.isFirstOfPair() ? 1 : 2) : OptionalInt.empty(),
+                 read.isReverseStrand() ? BaseUtils.simpleReverseComplement(read.getBases()) : read.getBases(),
+                 reverseIf(read.isReverseStrand(), read.getBaseQualities()), new Mapping(read));
```

Comment:
My comment pertained to the ```new Mapping(read)``` part of the line.  Since you allowed for a null Mapping object in composeHeaderLine, I thought you'd want to pass null for an unmapped read.  The way you have it, you'll produce a Mapping with contig=null, start = -1, strand = +, and cigar = null.  This will cause Mapping::toString to blow up, because you don't account for null cigars.